extensions/spring-web/deployment/src/main/java/io/quarkus/spring/web/deployment/ControllerAdviceAbstractExceptionMapperGenerator.java

```diff
@@ -135,17 +137,39 @@ void generateMethodBody(MethodCreator toResponse) {
 
             ResultHandle response;
             if (RESPONSE_ENTITY.equals(returnType.name())) {
+                /*
+                 * By default we will send JSON back unless the ResponseEntity has a String body
+                 */
+                boolean addDefaultJsonContentType = true;
+                if (returnType.kind() == Type.Kind.PARAMETERIZED_TYPE) {
+                    if (returnType.asParameterizedType().arguments().size() == 1) {
+                        Type responseEntityParameterType = returnType.asParameterizedType().arguments().get(0);
+                        if (STRING.equals(responseEntityParameterType.name())) {
+                            addDefaultJsonContentType = false;
```

Comment:
There's no way in Spring to explicitly define the response media type?

Reply:
For the regular controller method sure there is and we handle it. But for error handler you either return an object and rely on the default or return a `ResponseEntity` where you can set whatever you like.

Reply:
Are you sure about that? Wouldn't it use the content type defined in the `@RequestMapping` annotation of the original method?

And just to be clear, we don't support overriding the content-type in the `ResponseEntity` response of the exception mapper?

Reply:
> Are you sure about that? Wouldn't it use the content type defined in the `@RequestMapping` annotation of the original method?
> 

Yes it would, and that is what causes the problem, since the controller method uses a type that doesn't match what the exception handler returns.

> And just to be clear, we don't support overriding the content-type in the `ResponseEntity` response of the exception mapper?

That's part of the responsibility of the application code. If the application sets it, we don't change it: https://github.com/quarkusio/quarkus/blob/63673a7189713c822366b518550248778ab4a66d/extensions/spring-web/runtime/src/main/java/io/quarkus/spring/web/runtime/ResponseEntityConverter.java#L36

Reply:
We discussed this on Zulip with @geoand. Our current support is a bit limited as we will always return JSON (except in some edge cases), even if the original method was pushing some XML.

That's not very user friendly and we will probably need to improve that. But that can probably wait so we decided to merge that one first.

Reply:
As discussed I will research this a bit more and open an issue for a follow-up implementation